drivers/serial/uart_mcux_lpuart.c

```diff
@@ -970,9 +974,47 @@ static void mcux_lpuart_isr(const struct device *dev)
 }
 #endif /* CONFIG_UART_MCUX_LPUART_ISR_SUPPORT */
 
+#if LPUART_HAS_MODEM
+static int mcux_lpuart_config_flowctrl(uint8_t flow_ctrl, lpuart_config_t *uart_config)
+{
+	switch (flow_ctrl) {
+	case UART_CFG_FLOW_CTRL_NONE:
+		uart_config->enableTxCTS = false;
+		uart_config->enableRxRTS = false;
+		break;
+
+	case UART_CFG_FLOW_CTRL_RS485:
+		uart_config->enableTxCTS = false;
+		uart_config->enableRxRTS = false;
+		break;
+
+	case UART_CFG_FLOW_CTRL_RTS_CTS:
+		uart_config->enableTxCTS = true;
+		uart_config->enableRxRTS = true;
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+#else
+static int mcux_lpuart_config_flowctrl(uint8_t flow_ctrl, lpuart_config_t *uart_config)
+{
+	if (flow_ctrl == UART_CFG_FLOW_CTRL_NONE) {
+		return 0;
+	}
+
+	return -ENOTSUP;
```

Comment:
very nit but would prefer to have the

```
if (flow_ctrl != ...NONE) {
  return -ENOTSUP
}

return 0;
```
just because thats the normal paradigm is to check errors and then last return 0. right now not a big deal but if function gets bigger this is easier to edit

Reply:
makes sense, updated